src/components/Header/Header.tsx

```diff
@@ -0,0 +1,67 @@
+import { useContext, useState } from 'react';
+import { AuthContext } from '../Auth/AuthContext';
+import { postTodo } from '../../api/todos';
+
+import { Error } from '../../types/Error';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  setTodos: (arg: Todo[]) => void,
+  setIsError: (arg: Error | null) => void,
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setIsError,
+}) => {
+  const [query, setQuery] = useState('');
+  const user = useContext(AuthContext);
+
+  const postTodosData = async () => {
+    if (user && query) {
+      const data = {
+        title: query,
+        userId: user.id,
+        completed: false,
+      };
+
+      await postTodo(data)
+        .then(todo => {
+          if (todos) {
```

Comment:
it is bad practice to use setter and getter in one place, and what difference does it make if you have todos or not?